lib/kcerts/ssh.go

```diff
@@ -100,6 +106,51 @@ func (a SSHAgent) Valid() bool {
 	return err == nil
 }
 
+func (a *SSHAgent) UseStandardPaths() error {
+	user, err := UserCurrent()
+	if err != nil {
+		return err
+	}
+
+	// Make sure /tmp/ssh-${USER} exists with the right permissions
+	path := fmt.Sprintf("/tmp/ssh-%s", user.Username)
+	socket := filepath.Join(path, "agent")
+
+	if a.Socket == socket {
+		// no standardization needed.
+		return nil
+	}
+
+	// Securely make sure path exists and is permission 0700
+	// TODO(jonathan): Would it be safer to destroy and remake this directory?
+	if err := os.Mkdir(path, 0700); err != nil && !os.IsExist(err) {
+		return err
+	}
+	if err := os.Chmod(path, 0700); err != nil {
+		return err
+	}
+
+	// Securely rename socket:
+	_, err = os.Stat(socket)
+	if err == nil {
+		err := os.Remove(socket)
+		if err != nil {
+			return err
+		}
+	} else if !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
+	if err := os.Chmod(a.Socket, 0700); err != nil {
+		return err
+	}
+	if err := os.Link(a.Socket, socket); err != nil {
```

Comment:
uhm .. this would fail on many many systems? hard links don't work across filesystems, /tmp is often a tmpfs, likely not on the same filesystem where ssh-agent creates the socket by default.

Reply:
I thought I tried a symlink and had an issue, but maybe there was a user error in there.  I'll try again and see!  symlinks are definitely easier to inspect than hardlinks.

Reply:
Now using a symlink.